### PR TITLE
fix: complete copyWithZone for TorrentCell

### LINKED ISSUE #5077: Transmission v4.0.1 crashes on startup when launching app
### What is the issue?

I had installed nightly build on macOS 13.2.1. macOS 13.2.1 has two user accounts. One account has not been used with Transmission. Second account has transmissions not finished by Transmission nightly build. Nightly build has been uninstalled and replaced by Transmission v4.0.1.

Transmission v.4.0.1 can be launched and is starting on account with no transmissions, but on account with unfinished transmissions app crashes on startup.

How to migrate from nightly build to v.4.0.1 reusing unfinished transmissions?

[crash_Transmission.pdf](https://github.com/transmission/transmission/files/10839908/crash_Transmission.pdf)

### Which application of Transmission?

macOS app

### Which version of Transmission?

v4.0.1

### PR DESCRIPTION
**Pull request by Coeur**
This is a best guess attempt to fix #5077.
I do not know if it will have any impact for @AWHubGit. The best thing would be for them to try this change and give us feedback if it had any impact.

### CODE REVIEW

File: macosx/TorrentCell.mm [hunk 1]
```diff
@@ -101,9 +99,19 @@ - (instancetype)init
 
 - (id)copyWithZone:(NSZone*)zone
 {
-    id value = [super copyWithZone:zone];
-    [value setRepresentedObject:self.representedObject];
-    return value;
+    TorrentCell* copy = [super copyWithZone:zone];
+    copy->_fTitleAttributes = [_fTitleAttributes mutableCopyWithZone:zone];
+    copy->_fStatusAttributes = [_fStatusAttributes mutableCopyWithZone:zone];
```
Comment (Coeur) [inline macosx/TorrentCell.mm:104]: This is the core of the fix attempt: adding `mutablecopywithzone:` to those two mutable properties.

**Comment by ckerr** [comment]
@AWHubGit when CI finishes, a macOS build will be available at https://github.com/transmission/transmission/actions/runs/4285064883 if you scroll down to the "Assets" section

**Comment by AWHubGit** [comment]
TransmissionArtifact.app is in /Applications side-by-side with Transmission.app (v4.0.1):

`xx@imacos /Applications % ls -all Transmission*
Transmission.app:
total 0
drwxr-xr-x@   3 aw    admin    96 22 Feb 23:56 .
drwxrwxr-x  103 root  admin  3296 28 Feb 09:49 ..
drwxr-xr-x@   9 aw    admin   288 22 Feb 23:56 Contents

TransmissionArtifact.app:
total 0
drwxrwxr-x    3 aw    staff    96 27 Feb 17:29 .
drwxrwxr-x  103 root  admin  3296 28 Feb 09:49 ..
drwxrwxr-x    8 aw    staff   256 27 Feb 17:29 Contents
aw@imacos12 /Applications %`

`xx@imacos /Applications % xattr -l Transmission.app
com.apple.macl: �l���I����Lt�f!��F@�X!^۰S����[�N�<@h����Yݣ�B���I��>�
                                                                    ���AO���£4d��W�0BCk���/y��
com.apple.provenance:
com.apple.quarantine: 01c3;63fa24ef;Safari;145BB345-68D0-41A2-A6B6-D865EA1FEDB3`

TransmissionArtifact.app no extended attributes.

"codesign --test-requirement="=notarized" --verify --verbose Transmission.app"
results in "valid on disk", "satifies its Designated Requirement" and "explicit requirement satisfied".

Same for TransmissionArtifact.app results in:
"invalid signature" and "In architecture: x86_64"

I haven't copied dir struct  /share/doc

TransmissionArtifact.app doesn't launch. "Can't be opened".

How to get TransmisionArtifact.app notarized?

**Comment by GaryElshaw** [comment]
ctrl-click (right-click) and choose 'open' should work.

**Comment by AWHubGit** [comment]
> ctrl-click (right-click) and choose 'open' should work.

`xx@imacos /Applications % open TransmissionArtifact.app
The application cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x600000ca6010 {Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}}
`

**Comment by nevack** [comment]
> > ctrl-click (right-click) and choose 'open' should work.
> 
> `xx@imacos /Applications % open TransmissionArtifact.app The application cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x600000ca6010 {Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}} `

You need to Open it with Finder.
https://support.apple.com/en-us/HT202491

Or you can remove quarantine flag with:
```bash
xattr -d com.apple.quarantine /path/to/Transmission.app
```

Or maybe with `-r` flag:
```bash
xattr -rd com.apple.quarantine /path/to/Transmission.app
```

**Comment by AWHubGit** [comment]
> > > ctrl-click (right-click) and choose 'open' should work.
> > 
> > 
> > `xx@imacos /Applications % open TransmissionArtifact.app The application cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x600000ca6010 {Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}} `
> 
> You need to Open it with Finder. https://support.apple.com/en-us/HT202491
> 
> Or you can remove quarantine flag with:
> 
> ```shell
> xattr -d com.apple.quarantine /path/to/Transmission.app
> ```
> 
> Or maybe with `-r` flag:
> 
> ```shell
> xattr -rd com.apple.quarantine /path/to/Transmission.app
> ```

xx@imacos /Applications % sudo xattr -rd com.apple.quarantine TransmissionArtifact.app
Password:
xx@imacos /Applications % sudo xattr -rd com.apple.quarantine ./TransmissionArtifact.app
xx@imacos /Applications % open TransmissionArtifact.app
The application cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x6000036a93e0 {Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}}
xx@imacos /Applications % open -a TransmissionArtifact.app
The application /Applications/TransmissionArtifact.app cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x6000037888d0 {Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}}

"sudo xattr -rd com.apple.quarantine TransmissionArtifact.app" exits with zero status means success.

**Comment by Coeur** [comment]
Same issue for me:
In binaries-macos-11, we have an Applications folder, and inside a Transmission.app of 20.6 MB.
But I'm unable to open it: it says "Impossible to open" and that's all.

**Comment by Coeur** [comment]
@nevack can you review the code change and see if it's safe and worth merging anyway?

**Review by DevilDimon — APPROVED** [review]

**Comment by Coeur** [comment]
@ckerr feel free to merge: I don't think it can cause any regression. And then AWHubGit will be able to test it more easily.

**Comment by nevack** [comment]
> @nevack can you review the code change and see if it's safe and worth merging anyway?

Sorry, but i'm not that familiar with this part of ObjC runtime.
I gotta trust you and @DevilDimon here.

**Comment by ckerr** [comment]
@AWHubGit according to [this page](https://developer.apple.com/forums/thread/676627) you may be able to resolve the `Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111"` issue with this:

```sh
$ chmod +x AppName.app/Contents/MacOS/AppName
```

**Comment by Coeur** [comment]
From the build after rebase:
https://github.com/transmission/transmission/actions/runs/4300882577

I did @ckerr command:
```
chmod +x Transmission.app/Contents/MacOS/Transmission 
```

But opening is still impossible. It might not be a build compatible with Apple Silicon? I believe the options left are either building from source, or merging the code and waiting from a proper universal build.

**Comment by willsaccount** [comment]
On mac m1 I receive this message attempting to open the build: 

“Transmission” is damaged and can’t be opened. You should move it to the Bin.

**Comment by AWHubGit** [comment]
> @AWHubGit according to [this page](https://developer.apple.com/forums/thread/676627) you may be able to resolve the `Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111"` issue with this:
> 
> ```shell
> $ chmod +x AppName.app/Contents/MacOS/AppName
> ```

I made all files in "/Applications/TransmissionArtifact.app/Contents/MacOS" executable, but opening is still impossible.

Attached logs

- "ls -allR /Applications/Transmission.app > /Users/xx/Desktop/ls_allR_Transmission.app_2.log"
- "ls -allR /Applications/TransmissionArtifact.app > /Users/xx/Desktop/ls_allR_TransmissionArtifact.app_2.log"

Install https://formulae.brew.sh/cask/kdiff3 [https://formulae.brew.sh/cask/kdiff3](https://formulae.brew.sh/cask/kdiff3) on your system. It breaks down diffs down to single characters - not only lines. Great tool.

Then you can see all diffs between these two branches (Transmission v4.0.1 and https://github.com/transmission/transmission/actions/runs/4310613633 installed by me as TransmissionArtifact).

[ls_allR_Transmission.app_2.log](https://github.com/transmission/transmission/files/10869390/ls_allR_Transmission.app_2.log)
[ls_allR_TransmissionArtifact.app_2.log](https://github.com/transmission/transmission/files/10869391/ls_allR_TransmissionArtifact.app_2.log)

**Comment by GaryElshaw** [comment]
I'm pretty confident this is related to SIP protection and signing. Another thing to try if you feel up to it @AWHubGit. https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection

**Comment by nevack** [comment]
> I'm pretty confident this is related to SIP protection and signing. Another thing to try if you feel up to it @AWHubGit. [developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection](https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection)

I won't suggest anyone disabling SIP.
It is important security feature, not that easy to manage correctly, and can leave user vulnerable and without Touch ID/Apple Pay and other related protected features.

**Comment by ckerr** [comment]
What is the right way to move forward on this and let this PR get testing?

**Comment by GaryElshaw** [comment]
I think it has to be merged.

**Comment by Coeur** [comment]
@ckerr I tested it by building from sources, and it doesn't cause regressions in performances or anything. So it should be good to merge and then we'll get the nightly build which works for Apple Silicon.

**Comment by ckerr** [comment]
This particular PR looks fine so I'm OK with merging it -- but a workflow of "we have to merge it before it can be tested on Silicon" isn't great. I wonder if there's something we can do to improve that.

**Comment by GaryElshaw** [comment]
I have been able to run things from CI before, but my setup will no doubt be different. It would be great if CI and nightlies could be bridged somehow, is there any way to change the CI binary output?

**Comment by nevack** [comment]
> I wonder if there's something we can do to improve that.

I think it's fixable but current gigantic workflows are PITA to work with 😄

**Comment by ckerr** [comment]
Notes: Fixed crash in `copyWithZone()`.

**Pull request merged**